chart/templates/controller.yaml

```diff
@@ -72,34 +72,45 @@ data:
     #!/bin/env bash
     # Get the in-cluster kubeconfig for KubeFlex Control Planes
     # get-kubeconfig.sh cp_name guess_its_name
+
+    # input parameters
     cp_name="${1%"-system"}" # cp name or cp namespace
-    guess_its_name="$2" # true: try guessing the name of the ibms CP
-    if [ "$cp_name" == "" ] ; then
+    guess_its_name="$2" # true: try guessing the name of the ITS CP
+
+    # check if the CP name is valid or needs to be guessed
+    while [ "$cp_name" == "" ] ; do
       if [ "$guess_its_name" == "true" ] ; then
-        for cp in `kubectl get controlplane -o name`; do
-          cp=${cp##*/} # separate just the CP name
-          if [ "$(kubectl get controlplane $cp -o jsonpath="{.metadata.labels['kflex\.kubestellar\.io/cptype']}")" == "its" ] ; then
-            if [ "$cp_name" == "" ] ; then
-              cp_name="$cp"
-            else
-              >&2 echo "ERROR: found more than one Control Plane of type its!"
-              exit 1
-            fi
-          fi
-        done
-        if [ "$cp_name" == "" ] ; then
-          >&2 echo "ERROR: no Control Plane of type its found!"
-          exit 2
-        fi
+        cps=$(kubectl get controlplane -l 'kflex.kubestellar.io/cptype=its' 2> /dev/null | tail -n +2)
+        case $(echo -n "$cps" | grep -c '^') in

```

Comment:
FYI, `wc -l` is a thing.

Reply:
I used `wc -l` initially but it does not deal with an empty variable well. I found the grep based solution online to solve the problem

Reply:
I am not sure what you mean by "an empty variable".

```
bash-3.2$ echo -n "" | wc -l
       0
bash-3.2$ echo -n "" | grep -c "^"
0
```

Working code is good enough.